p2p/swarm.go

```diff
@@ -284,6 +284,32 @@ func (s *swarm) SendMessage(peerPubkey p2pcrypto.PublicKey, protocol string, pay
 	return s.sendMessageImpl(peerPubkey, protocol, service.DataBytes{Payload: payload})
 }
 
+func (s *swarm) RetrySend(peerPubKey p2pcrypto.PublicKey, message []byte) error {
+	var err error
+	var peer node.Node
+	var conn net.Connection
+
+	peer, err = s.dht.Lookup(peerPubKey) // blocking, might issue a network lookup that'll take time.
```

Comment:
please use the latest code from `develop`. you copied this from an older version of `sendMessageImpl`.
also it'll be better if you take out the code that tries to achieve a connection to a function and use it in both places. will reduce some code duplication and will save us from errors like the one I've mentioned above in the future.